vllm/config.py

```diff
@@ -543,6 +564,11 @@ def contains_seqlen_agnostic_layers(
     def get_layers_block_type(self,
                               parallel_config: "ParallelConfig") -> List[str]:
         num_layers = self.get_num_layers(parallel_config)
+
+        if self.is_attention_free:
+            assert (self.hf_config.model_type == "mamba")
+            return ["mamba"] * num_layers
```

Comment:
Maybe you could look for the presence of one of these timestep keys, or at the very least we should pull out this check into a list of archs
https://huggingface.co/mistralai/Mamba-Codestral-7B-v0.1/blob/88085f9cdfa832c3aca8a0315a4520cf7558c947/config.json#L28-L37
https://huggingface.co/state-spaces/mamba-130m-hf/blob/1e76775f628fbf1350fbe4dbb3d971ba64af25a1/config.json#L27-L32